multitemporal/mt.py

```diff
@@ -316,17 +325,26 @@ def run(projdir, outdir, projname, sources, steps,
                         'specified input: {}\n\tpossible inputs: {}'
                         .format(thisinput, [s['name'] for s in steps]))
                 thisnin = parentsteps[0]['nout']
+                thisnyrin = parentsteps[0]['nyrout']
             if 'nin' in step:
                 assert step['nin'] == thisnin, "Number of inputs do not match"
             else:
                 step['nin'] = thisnin
+            if 'nyrin' in step:
+                assert step['nyrin'] == thisnyrin, "Number of years do not match"
```

Comment:
What would happen if the user disabled assertions when running this?  That's how to tell if it should be a `raise ValueError` instead.